scripts/dll/build-dlls.js

```diff
@@ -7,53 +7,132 @@
 
 /* eslint-env node */
 
+const chalk = require( 'chalk' );
 const childProcess = require( 'child_process' );
+const minimist = require( 'minimist' );
 const path = require( 'path' );
-const chalk = require( 'chalk' );
 
-const ROOT_DIRECTORY = path.resolve( __dirname, '..', '..' );
-const IS_DEVELOPMENT_MODE = process.argv.includes( '--dev' );
+const argv = minimist( process.argv.slice( 2 ) );
 
-if ( IS_DEVELOPMENT_MODE ) {
-	console.log( '🛠️️  ' + chalk.yellow( 'Development mode is active.' ) );
-} else {
-	console.log( '⚠️  ' + chalk.magenta( 'Production mode is active. Use --dev to build in the development mode.' ) );
-}
+const ROOT_DIRECTORY = argv.cwd ? path.resolve( argv.cwd ) : path.resolve( __dirname, '..', '..' );
+const BASE_DLL_CONFIG_PATH = argv[ 'base-dll-config' ] ? path.relative( ROOT_DIRECTORY, argv[ 'base-dll-config' ] ) : null;
+const BASE_DLL_PATH = argv[ 'base-dll-path' ];
+const IS_DEVELOPMENT_MODE = argv.dev;
+const VERBOSE_MODE = argv.verbose;
+
+// Lets highlight and space out the messages in the verbose mode to
+// make them stand out from the wall of text that webpack spits out.
+const prefix = VERBOSE_MODE ? '\n📍 ' : '';
 
-// -------------------------------------------------------------
-// ------------------------------------------- Base DLL build --
+if ( BASE_DLL_CONFIG_PATH ) {
+	console.log( prefix + chalk.bold( 'Creating the base DLL build...' ) );
 
-console.log( '\n📍 ' + chalk.cyan.underline( 'Creating the base DLL build...\n' ) );
+	const status = execute( {
+		command: [ 'yarn', 'webpack', `--config=${ normalizePath( BASE_DLL_CONFIG_PATH ) }` ],
+		cwd: BASE_DLL_PATH || ROOT_DIRECTORY
+	} );
 
-const webpackArguments = [ '--config=./scripts/dll/webpack.config.dll.js' ];
+	if ( status ) {
+		console.log( chalk.bold.red( 'Halting the script due to failed base DLL build.' ) );
 
-if ( IS_DEVELOPMENT_MODE ) {
-	webpackArguments.push( '--mode=development' );
+		process.exit();
```

Comment:
If an error occurs, we should set the exit code to a non-zero value.

![image](https://user-images.githubusercontent.com/2270764/194227429-e7518e0f-384a-458c-9fd0-528b9b3bb701.png)